client.go

```diff
@@ -127,6 +132,41 @@ func (c *Client) recvVersion() error {
 	return nil
 }
 
+func (c *Client) recv() {
+	sendAll := func(res result) {
+		c.mu.Lock()
+		listeners := make([]chan<- result, len(c.inflight))
+		for _, ch := range c.inflight {
+			listeners = append(listeners, ch)
+		}
+		c.mu.Unlock()
+		for _, ch := range listeners {
+			ch <- res
```

Comment:
The inflight channels are all controlled privately and some care is taken to ensure that all are drained in a timely manner. I don't believe this needs to be non-blocking, but I could make an initial non-blocking pass through the listeners and followup with a blocking pass to catch all the remaining if you prefer.